keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -323,11 +338,180 @@ myOtherField: newly added field
 The feature proposed in this KEP would allow this change, since `myField` was
 unaltered from the original object. If `myField` was changed, the new value would have to pass the new validation rule to be written.
 
+### Nested Value Validations 
+
+OpenAPI-schemas support nested validations with logical operations:
+- not
+- oneOf
+- anyOf
+- allOf
+
+To keep the semantics simple and implementation clear, schemas and validations
+nested within the following will be not ratcheted:
+
+- not
+- oneOf
+- anyOf
+
+Our intuition is that in the wild CRD authors do not make much use of them, and
+when they do the logic is complicated and references multiple fields. 
+
+For all these constructs, the conservative `DeepEqual` rule still applies 
+to the final output of the rule. 
+
+The following will be ratcheted with the conservative ratcheting rule, and
+their children will also be ratcheted.
+
+- allOf
```

Comment:
I wasn't clear to me just from reading this section of the KEP why we are including `allOf`.  Can we clarify?

Reply:
Added a justification for why `allOf` can be safely ratcheted with `DeepEqual`. Not sure if this bit will make the final implementation, but thought it would be good to leave the door open if we can

Reply:
Ah, the problem is ambiguity that is created when multiple validations combined are ratcheting. That makes sense.